### PR TITLE
fix changging the stopped vm's subnets, the vm cann't start normally

### LINKED ISSUE #2461: if we change the vm's subbnets when vm is stopped and keepping vm ip ,we start the vm,  the launcher pod will report error
# Expected Behavior
vm start normally 
# Actual Behavior
vm start unnormally and the  launcher pod  will report ovn-nbctl: vm-xxx.default.attachnet.default.ovn: port already exists but in switch dual
# Steps to Reproduce the Problem

1. kubectl edit vm 
1. virtctl start vm
1.

# Additional Info

- Kubernetes version:

  **Output of `kubectl version`:**

  ```
  v1.21.5
  ```

- kube-ovn version:
  
  ```
  v1.12.0
  ```

- operation-system/kernel version:
  
  **Output of `awk -F '=' '/PRETTY_NAME/ { print $2 }' /etc/os-release`:**
  **Output of `uname -r`:**
  ```
  (paste your output here)
  ```

### PR DESCRIPTION
**Pull request by lynn901**
- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

#### What type of this PR
Examples of user facing changes:
- Bug fixes 

#### Which issue(s) this PR fixes:
Fixes #2461

### CODE REVIEW

File: pkg/controller/pod.go [hunk 1]
```diff
@@ -604,6 +604,38 @@ func (c *Controller) handleAddPod(key string) error {
...
+			if ipCr != nil {
+				if ipCr.Spec.Subnet != subnet.Name {
+					key := fmt.Sprintf("%s/%s", pod.Namespace, vmName)
+					ports, err := c.ovnClient.ListNormalLogicalSwitchPorts(true, map[string]string{"pod": key})
+					if err != nil {
+						klog.Errorf("failed to list lsps of pod '%s', %v", pod.Name, err)
+						return err
+					}
+					for _, port := range ports {
+						if err != nil {
```
Comment (hongzhen-ma) [inline pkg/controller/pod.go:627]: what's this err used for?

Comment (hongzhen-ma) [inline pkg/controller/pod.go:627]: what's this err used for?

File: pkg/controller/pod.go [hunk 2]
```diff
@@ -604,6 +604,38 @@ func (c *Controller) handleAddPod(key string) error {
...
+						}
+						// when lsp is deleted, the port of pod is deleted from any port-group automatically.
+						klog.Infof("gc logical switch port %s", port.Name)
+						if err := c.ovnLegacyClient.DeleteLogicalSwitchPort(port.Name); err != nil {
+							klog.Errorf("failed to delete lsp %s, %v", port.Name, err)
+							return err
+						}
+					}
+				}
+			}
```
Comment (hongzhen-ma) [inline pkg/controller/pod.go:638]: Is it possible to abstract the process as a function and called in  handleAddPod?

Comment (lynn901) [reply pkg/controller/pod.go:635]: > Is it possible to abstract the process as a function and called in handleAddPod?
it sounds good

Comment (lynn901) [reply pkg/controller/pod.go:635]: > Is it possible to abstract the process as a function and called in handleAddPod?

it sounds ok，i will change it

**Review by hongzhen-ma — APPROVED** [review]

**Comment by hongzhen-ma** [comment]
已回合release-1.11 release-1.10 和 release-1.9 分支

**Pull request merged**